src/Bicep.Core/LanguageConstants.cs

```diff
@@ -151,14 +153,14 @@ public static IEnumerable<TypeProperty> GetCommonResourceProperties(ResourceType
             yield return new TypeProperty(ResourceApiVersionPropertyName, new StringLiteralType(reference.ApiVersion), TypePropertyFlags.ReadOnly | TypePropertyFlags.DeployTimeConstant);
         }
 
-        private static ResourceReferenceType CreateResourceScopeReference(ResourceScope resourceScope)
+        private static ResourceScopeType CreateResourceScopeReference(ResourceScope resourceScope)
             => resourceScope switch
             {
-                ResourceScope.Tenant => new ResourceReferenceType("tenant", resourceScope),
-                ResourceScope.ManagementGroup => new ResourceReferenceType("managementGroup", resourceScope),
-                ResourceScope.Subscription => new ResourceReferenceType("subscription", resourceScope),
-                ResourceScope.ResourceGroup => new ResourceReferenceType("resourceGroup", resourceScope),
-                _ => new ResourceReferenceType("none", ResourceScope.None),
+                ResourceScope.Tenant => new ResourceScopeType("tenant", resourceScope),
+                ResourceScope.ManagementGroup => new ResourceScopeType("managementGroup", resourceScope),
+                ResourceScope.Subscription => new ResourceScopeType("subscription", resourceScope),
+                ResourceScope.ResourceGroup => new ResourceScopeType("resourceGroup", resourceScope),
+                _ => new ResourceScopeType("none", ResourceScope.None),
```

Comment:
>ResourceScopeType [](start = 25, length = 17)

Just out of curiosity - this means that no resource can deploy to this scope?

Reply:
Yes. This block is only used to generate scope types for modules; technically it's not possible to declare any other scope, but the exhaustive pattern matching necessitates this. I'll see if I can restructure to make this more intuitive.